### PR TITLE
Users can view existing expenses

### PR DESCRIPTION
**Pull request by hrs**
* Add an Expense model and factory.
* Expenses belong to a User and a Category.
* Expenses have a name and amount.
* Add an expenses#show page.

### CODE REVIEW

File: app/controllers/expenses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class ExpensesController < ApplicationController
+  def show
+    @expense = find_expense
+  end
+
+  private
+
+  def find_expense
```
Comment (jencheng) [inline app/controllers/expenses_controller.rb]: interesting you puled this out into its own method

File: db/migrate/20151027025641_create_expenses.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class CreateExpenses < ActiveRecord::Migration
+  def change
+    create_table :expenses do |t|
+      t.string :name, null: false
+      t.decimal :amount, null: false, default: 0.0
```
Comment (jencheng) [inline db/migrate/20151027025641_create_expenses.rb]: amount is `float`? are you worried about imprecision?

File: app/controllers/expenses_controller.rb (hunk 1 above)
Comment (hrs) [inline app/controllers/expenses_controller.rb]: I usually pull that pattern into its own method. It tends to get used in other actions, and I just find it a little easier to read anyway.

File: db/migrate/20151027025641_create_expenses.rb (hunk 2 above)
Comment (hrs) [inline db/migrate/20151027025641_create_expenses.rb]: Not to within two decimal points (and we're not a bank, anyway). Do you think it's risky?

**Comment by jencheng** [comment]
looks good!
some questions to consider:
* when an expense is `shared`, what if the $ amount doesn't divide nicely? say, `3 cents`
* maybe storing things as `ints` instead of `floats` for better precision? you can covert for display

File: db/migrate/20151027025641_create_expenses.rb (hunk 2 above)
Comment (jencheng) [inline db/migrate/20151027025641_create_expenses.rb]: I've just normally seen $ handled as ints. No big deal here.

**Comment by hrs** [comment]
@jencheng At the moment there's no logic for sharing; it's just a boolean field. However, I imagine that when you split $0.03 you'll just say that each person owes $0.015.

I'm generally not too worried about perfect precision. If we lose a cent here or there, that's really not a problem. We're usually more interested in dollars, anyway. =)

**Pull request merged**